To diagnose problem with truncated left and right files:


    tail /home/ljcohen/osmotic_assemblies_completed/F_zebrinus/F_zebrinus.right.fq
    tail /home/ljcohen/osmotic_assemblies_completed/F_zebrinus/F_zebrinus.left.fq
    cd /home/ljcohen/osmotic_assemblies_completed/F_zebrinus/




    cat *.2 > test.right.fq
    tail test.right.fq 

This is normal! Run another one, if that is normal. Therefore, problem with cat.

Pick L_goodei.
    cd /home/ljcohen/osmotic_assemblies_completed/L_goodei/ 
    tail L_goodei.right.fq 
    tail L_goodei.left.fq 

These look normal! Run snap aligner to see if it works on L_goodei.

    screen
    srun --mem 32GB --pty bash
    source ~/.bashrc
    cd ~/test
    snap-aligner index /home/ljcohen/msu_assemblies_finished/L_goodei/L_goodei.Trinity.fixed.fa .

Output:

    55986941(12%) seeds occur more than once, total of 183518071(40%) genome locations are not unique, 167901523(37%) bad seeds, 0 both complements used 6730000 no string

Then

    /home/ljcohen/bin/snap-aligner paired \
    /home/ljcohen/test \
    /home/ljcohen/osmotic_assemblies_completed/L_goodei/L_goodei.left.fq /home/ljcohen/osmotic_assemblies_completed/L_goodei/L_goodei.right.fq  -o L_goodei.bam \
    -s 0 1000 -H 300000 -h 2000 -d 30 -t 20 -b -M -D 5 -om 5 -omax 10 -mcp 99999999

But then this happened :(

    FASTQ file doesn't end with a newline!  Failing.  fileOffset = 5526127364, validBytes = 60
    SNAP exited with exit code 1 from line 252 of file SNAPLib/FASTQ.cpp

Then, went to msu.hpcc and remembered that reads used for assemblies were actually fasta files, from before we discovered the error from a discrepency between screed khmer versions. So, assemblies might not be screwed up after all? 

Combine A_xenica:

    cat *.2 > A_xenica.right.test.fq
    cat *.1 > A_xenica.left.test.fq

Do this for F_zebrinus also. Transfer to msu, assemble. Then, compare assemblies to see how different.
