<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Strict//EN">
<html>
<head><title>F_similis_NA_FW_1_R1.fastq.gz FastQC Report</title>

<style type="text/css">

 @media screen {
  div.summary {
    width: 18em;
    position:fixed;
    top: 3em;
    margin:1em 0 0 1em;
  }
  
  div.main {
    display:block;
    position:absolute;
    overflow:auto;
    height:auto;
    width:auto;
    top:4.5em;
    bottom:2.3em;
    left:18em;
    right:0;
    border-left: 1px solid #CCC;
    padding:0 0 0 1em;
    background-color: white;
    z-index:1;
  }
  
  div.header {
    background-color: #EEE;
    border:0;
    margin:0;
    padding: 0.5em;
    font-size: 200%;
    font-weight: bold;
    position:fixed;
    width:100%;
    top:0;
    left:0;
    z-index:2;
  }

  div.footer {
    background-color: #EEE;
    border:0;
    margin:0;
	padding:0.5em;
    height: 1.3em;
	overflow:hidden;
    font-size: 100%;
    font-weight: bold;
    position:fixed;
    bottom:0;
    width:100%;
    z-index:2;
  }
  
  img.indented {
    margin-left: 3em;
  }
 }
 
 @media print {
	img {
		max-width:100% !important;
		page-break-inside: avoid;
	}
	h2, h3 {
		page-break-after: avoid;
	}
	div.header {
      background-color: #FFF;
    }
	
 }
 
 body {    
  font-family: sans-serif;   
  color: #000;   
  background-color: #FFF;
  border: 0;
  margin: 0;
  padding: 0;
  }
  
  div.header {
  border:0;
  margin:0;
  padding: 0.5em;
  font-size: 200%;
  font-weight: bold;
  width:100%;
  }    
  
  #header_title {
  display:inline-block;
  float:left;
  clear:left;
  }
  #header_filename {
  display:inline-block;
  float:right;
  clear:right;
  font-size: 50%;
  margin-right:2em;
  text-align: right;
  }

  div.header h3 {
  font-size: 50%;
  margin-bottom: 0;
  }
  
  div.summary ul {
  padding-left:0;
  list-style-type:none;
  }
  
  div.summary ul li img {
  margin-bottom:-0.5em;
  margin-top:0.5em;
  }
	  
  div.main {
  background-color: white;
  }
      
  div.module {
  padding-bottom:1.5em;
  padding-top:1.5em;
  }
	  
  div.footer {
  background-color: #EEE;
  border:0;
  margin:0;
  padding: 0.5em;
  font-size: 100%;
  font-weight: bold;
  width:100%;
  }


  a {
  color: #000080;
  }

  a:hover {
  color: #800000;
  }
      
  h2 {
  color: #800000;
  padding-bottom: 0;
  margin-bottom: 0;
  clear:left;
  }

  table { 
  margin-left: 3em;
  text-align: center;
  }
  
  th { 
  text-align: center;
  background-color: #000080;
  color: #FFF;
  padding: 0.4em;
  }      
  
  td { 
  font-family: monospace; 
  text-align: left;
  background-color: #EEE;
  color: #000;
  padding: 0.4em;
  }

  img {
  padding-top: 0;
  margin-top: 0;
  border-top: 0;
  }

  
  p {
  padding-top: 0;
  margin-top: 0;
  }
  
</style>

</head>
<body>
<div class="header">
<div id="header_title"><img src="Icons/fastqc_icon.png" alt="FastQC">FastQC Report</div>
<div id="header_filename">
Tue 2 Feb 2016<br />
F_similis_NA_FW_1_R1.fastq.gz
</div>
</div>
<div class="summary">
<h2>Summary</h2>
<ul>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M0">Basic Statistics</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M1">Per base sequence quality</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M2">Per sequence quality scores</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M3">Per base sequence content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M4">Per base GC content</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M5">Per sequence GC content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M6">Per base N content</a></li>
<li><img src="Icons/tick.png" alt="[PASS]"> <a href="#M7">Sequence Length Distribution</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M8">Sequence Duplication Levels</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M9">Overrepresented sequences</a></li>
<li><img src="Icons/error.png" alt="[FAIL]"> <a href="#M10">Kmer Content</a></li>
</ul>
</div>
<div class="main">
<div class="module"><h2 id="M0"><img src="Icons/tick.png" alt="[OK]"> Basic Statistics</h2>
<table>
<tr>
<th>Measure</th>
<th>Value</th>
</tr>
<tr>
<td>Filename</td>
<td>F_similis_NA_FW_1_R1.fastq.gz</td>
</tr>
<tr>
<td>File type</td>
<td>Conventional base calls</td>
</tr>
<tr>
<td>Encoding</td>
<td>Sanger / Illumina 1.9</td>
</tr>
<tr>
<td>Total Sequences</td>
<td>17019396</td>
</tr>
<tr>
<td>Filtered Sequences</td>
<td>0</td>
</tr>
<tr>
<td>Sequence length</td>
<td>100</td>
</tr>
<tr>
<td>%GC</td>
<td>46</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M1"><img src="Icons/tick.png" alt="[OK]"> Per base sequence quality</h2>
<p><img class="indented" src="Images/per_base_quality.png" alt="Per base quality graph"></p>
</div>
<div class="module"><h2 id="M2"><img src="Icons/tick.png" alt="[OK]"> Per sequence quality scores</h2>
<p><img class="indented" src="Images/per_sequence_quality.png" alt="Per Sequence quality graph"></p>
</div>
<div class="module"><h2 id="M3"><img src="Icons/error.png" alt="[FAIL]"> Per base sequence content</h2>
<p><img class="indented" src="Images/per_base_sequence_content.png" alt="Per base sequence content"></p>
</div>
<div class="module"><h2 id="M4"><img src="Icons/error.png" alt="[FAIL]"> Per base GC content</h2>
<p><img class="indented" src="Images/per_base_gc_content.png" alt="Per base GC content graph"></p>
</div>
<div class="module"><h2 id="M5"><img src="Icons/error.png" alt="[FAIL]"> Per sequence GC content</h2>
<p><img class="indented" src="Images/per_sequence_gc_content.png" alt="Per sequence GC content graph"></p>
</div>
<div class="module"><h2 id="M6"><img src="Icons/tick.png" alt="[OK]"> Per base N content</h2>
<p><img class="indented" src="Images/per_base_n_content.png" alt="N content graph"></p>
</div>
<div class="module"><h2 id="M7"><img src="Icons/tick.png" alt="[OK]"> Sequence Length Distribution</h2>
<p><img class="indented" src="Images/sequence_length_distribution.png" alt="Sequence length distribution"></p>
</div>
<div class="module"><h2 id="M8"><img src="Icons/error.png" alt="[FAIL]"> Sequence Duplication Levels</h2>
<p><img class="indented" src="Images/duplication_levels.png" alt="Duplication level graph"></p>
</div>
<div class="module"><h2 id="M9"><img src="Icons/error.png" alt="[FAIL]"> Overrepresented sequences</h2>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Percentage</th>
<th>Possible Source</th>
</tr>
<tr>
<td>TTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT</td>
<td>310315</td>
<td>1.8233020725294833</td>
<td>No Hit</td>
</tr>
<tr>
<td>GATCGGAAGAGCACACGTCTGAACTCCAGTCACTCCGCGAAATCTCGTAT</td>
<td>38168</td>
<td>0.224261777562494</td>
<td>TruSeq Adapter, Index 6 (97% over 36bp)</td>
</tr>
<tr>
<td>AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA</td>
<td>34139</td>
<td>0.20058878705213745</td>
<td>No Hit</td>
</tr>
</table>
</div>
<div class="module"><h2 id="M10"><img src="Icons/error.png" alt="[FAIL]"> Kmer Content</h2>
<p><img class="indented" src="Images/kmer_profiles.png" alt="Kmer graph"></p>
<table>
<tr>
<th>Sequence</th>
<th>Count</th>
<th>Obs/Exp Overall</th>
<th>Obs/Exp Max</th>
<th>Max Obs/Exp Position</th>
</tr>
<tr>
<td>TTTTT</td>
<td>54227730</td>
<td>20.558788</td>
<td>28.216904</td>
<td>10-14</td>
</tr>
<tr>
<td>AAAAA</td>
<td>23333685</td>
<td>11.671716</td>
<td>17.943287</td>
<td>95-96</td>
</tr>
<tr>
<td>GGGGG</td>
<td>2615115</td>
<td>2.5014515</td>
<td>5.9174213</td>
<td>1</td>
</tr>
<tr>
<td>CTCCA</td>
<td>3209055</td>
<td>2.2105787</td>
<td>5.8570027</td>
<td>1</td>
</tr>
<tr>
<td>GAAGA</td>
<td>3387625</td>
<td>2.1961734</td>
<td>5.2783175</td>
<td>6</td>
</tr>
<tr>
<td>CTCCT</td>
<td>3297440</td>
<td>2.1489718</td>
<td>5.6863365</td>
<td>1</td>
</tr>
<tr>
<td>CTGGA</td>
<td>3069790</td>
<td>2.1338103</td>
<td>6.130603</td>
<td>1</td>
</tr>
<tr>
<td>GGAGA</td>
<td>2848315</td>
<td>2.1021774</td>
<td>5.1232157</td>
<td>1</td>
</tr>
<tr>
<td>GAAAA</td>
<td>3666780</td>
<td>2.0880754</td>
<td>5.227918</td>
<td>2</td>
</tr>
<tr>
<td>GGAAG</td>
<td>2427255</td>
<td>1.7914172</td>
<td>5.3165417</td>
<td>5</td>
</tr>
<tr>
<td>GGAAA</td>
<td>2597060</td>
<td>1.6836559</td>
<td>6.3696003</td>
<td>1</td>
</tr>
<tr>
<td>GCCAG</td>
<td>1963715</td>
<td>1.6351271</td>
<td>5.1797194</td>
<td>1</td>
</tr>
<tr>
<td>CTCAG</td>
<td>2335840</td>
<td>1.6163338</td>
<td>5.2923484</td>
<td>1</td>
</tr>
<tr>
<td>GGGAG</td>
<td>1885955</td>
<td>1.5846109</td>
<td>5.354927</td>
<td>1</td>
</tr>
<tr>
<td>GTTTT</td>
<td>3435350</td>
<td>1.5672281</td>
<td>5.714783</td>
<td>1</td>
</tr>
<tr>
<td>CTGGG</td>
<td>1976300</td>
<td>1.5639037</td>
<td>5.301876</td>
<td>1</td>
</tr>
<tr>
<td>GAGCA</td>
<td>2089365</td>
<td>1.5350994</td>
<td>6.214093</td>
<td>9</td>
</tr>
<tr>
<td>AGAGC</td>
<td>2049245</td>
<td>1.5056224</td>
<td>5.9310546</td>
<td>8</td>
</tr>
<tr>
<td>GGGGA</td>
<td>1768570</td>
<td>1.4859821</td>
<td>5.643626</td>
<td>1</td>
</tr>
<tr>
<td>GGGAA</td>
<td>1925525</td>
<td>1.4211192</td>
<td>5.203587</td>
<td>1</td>
</tr>
</table>
</div>
</div><div class="footer">Produced by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a> (version 0.10.1)</div>
</body></html>