    cat /home/nreid/rnaseq/bwa.bams.list | grep "A_xenica"

slurm modules needed for Tophat:

    module load bowtie2/2.2.5 
    boost/1.55.0
    samtools/1.2
    tophat/2.1.0`

Tophat command:

    tophat -p 8 -o /home/ljcohen/test_tophat A_xenica_NA_BW_1.trim_1P.fq,A_xenica_NA_BW_2.trim_1P.fq  A_xenica_NA_BW_1.trim_2P.fq,A_xenica_NA_BW_2.trim_2P.fq,A_xenica_NA_BW_1.trim.orphans.fq.gz,A_xenica_NA_BW_2.trim.orphans.fq.gz

Run transrate with --left and --right reads. Snap still crashes. Tried this from [transrate google group](https://groups.google.com/forum/#!searchin/transrate-users/ran$20out$20of$20candidate$20pools$20-$20increase$20mcp/transrate-users/b7jKmPG5uos/0tf6sED3BwAJ)

    "You can manually edit the -mcp parameter. If you have the 'binary' download of transrate, you should edit the file ./lib/app/lib/transrate/snap.rb inside the transrate base directory.
    You'll need to edit the function `build_paired_cmd` - specifically you want to add a line. Find the line:

      > cmd << " -omax 10" # max alignments per pair/read

    And add a new line after it:

      > cmd << " -mcp 10000000" # maximum candidate pool size

    Save the file and re-run transrate."

Still didn't work. Install patch version of snap. Run interactively.

    srun --mem 32GB --pty bash

Index dnta fasta file first:

    /home/ljcohen/bin/snap-aligner index /home/ljcohen/msu_assemblies_finished/F_grandis/F_grandis.Trinity.fixed.fa .

Output:

    79369898(10%) seeds occur more than once, total of 259753154(34%) genome locations are not unique, 330751670(43%) bad seeds, 0 both complements used 13257060 no string

then run snap:

    /home/ljcohen/bin/snap-aligner paired \
    /home/ljcohen/msu_assemblies_finished/F_grandis/F_grandis.Trinity.fixed.fa \
    /home/ljcohen/osmotic_assemblies_completed/F_grandis/F_grandis.left.fq /home/ljcohen/osmotic_assemblies_completed/F_grandis/F_grandis.right.fq  -o test.bam \
    -s 0 1000 -H 300000 -h 2000 -d 30 -t 20 -b -M -D 5 -om 5 -omax 10 -mcp 99999999



    /home/ljcohen/bin/snap-aligner paired \
    /home/ljcohen/test \
    /home/ljcohen/osmotic_assemblies_completed/F_grandis/F_grandis.left.fq /home/ljcohen/osmotic_assemblies_completed/F_grandis/F_grandis.right.fq  -o test.bam \
    -s 0 1000 -H 300000 -h 2000 -d 30 -t 20 -b -M -D 5 -om 5 -omax 10 -mcp 99999999

Output: 

    FASTQ file doesn't end with a newline!  Failing.  fileOffset = 4547018492, validBytes = 260
    SNAP exited with exit code 1 from line 252 of file SNAPLib/FASTQ.cpp

This is the problem:

    ljcohen@c11-16:~/test$ tail /home/ljcohen/osmotic_assemblies_completed/F_grandis/F_grandis.left.fq
    +
    CCCFFFFFHHHGHIJIJJJJJJIJJJJJJIJJJJIJJJJIIGIIJJJIJJJJJJIJJJJJJIIIJJJJJJJJHIJFIHHHHHHHFFBDFEFEDEEECDDC
    @HWI-700460R:442:C5NU8ACXX:2:2214:13301:25421 1:N:0:GAGATTCCTATAGCCT
    CAAAGACTCGCTGAGCTTCAAGGTGTCGCATTTCCAGCAAATCAAACAGCCTCACACTTTGCAGGCAAAGACGTTGGAGGAGAAGAAGCTTTGGGCTCAT
    +
    CCCFFFFFHHHHHJJJJJJJJJJFGHHIJJJJJJJJJJJGIJJJJJJJJJEGIIIIGIJJGIJIHHHFBFFFDDDBDDDDBDDDDDDDDDDDDDDDDBC@
    @HWI-700460R:442:C5NU8ACXX:2:2214:13863:25278 1:N:0:GAGATTCCTATAGCCT
    GTCTGTGTGAGGTTTGAACTGGGAATTGCGAGGCACCAGAACTTTAGTGGAGCCCACCTTGTGCTGCTCCACCTCTGTGGCCCCGTCAAACTTTTCCTTG
    +
    CCCFFFFFHHHHHIJJJJJJJJJJJJJJIJJJJJJJJJJJJJJJJJJIJIJIJJJIJHHHHHFFFFFFEEEEEDDDDDDDDDDDDDDDljcohen@c11-16:~/test$

Try another set of reads that might not be truncated at the end. (Why did this happen? These files were made with khmer's split-paired-reads.py.)
