*Notes, 2016-02-12

"A basic recommendation is to have 1G of RAM per 1M pairs of Illumina reads."
- http://trinityrnaseq.github.io/performance/requirements.html

Need to think about monitoring resource use:
  - http://trinityrnaseq.github.io/performance/mem.html
  - http://trinityrnaseq.github.io/performance/
  - https://github.com/trinityrnaseq/trinityrnaseq/wiki/Trinity-Progress-Monitoring
  - sar: https://github.com/ctb/sartre/blob/master/README.txt

MSU cluster info:
  - https://wiki.hpcc.msu.edu/display/hpccdocs/HPCC+Quick+Reference+Sheet

Trinity reference:
  - https://github.com/trinityrnaseq/trinityrnaseq/wiki
  - http://trinityrnaseq.sourceforge.net/advanced_trinity_guide.html#Butterfly_reconstruction

Transrate:
  - http://hibberdlab.com/transrate/installation.html

BUSCO:
  - http://hibberdlab.com/transrate/installation.html

EviGene killifish RNA reference, use mRNAs:
  - http://arthropods.eugenes.org/EvidentialGene/killifish/RNAs/kf2evg367mixx11/

reciprocal best hit analysis:
  - http://khmer-protocols.readthedocs.org/en/ctb/mrnaseq/6-annotating-transcript-families.html

Diginorm streaming:
  - https://github.com/jessicamizzi/2015-ep-streaming
  - https://github.com/dib-lab/khmer-protocols/blob/jem-streaming/mrnaseq/1-quality.rst
  - https://github.com/dib-lab/dib-lab/issues/45#issuecomment-139825302

RNAseq transcriptome assembly references:
  - http://biorxiv.org/content/biorxiv/early/2015/12/30/035642.full.pdf
  - https://twitter.com/ctitusbrown/status/668865866944106496
  - http://genomebio.org/feeding-your-transcriptome-with-the-oyster-river-protocol/
